lib/src/main/java/io/ably/lib/util/AgentHeaderCreator.java

```diff
@@ -0,0 +1,61 @@
+package io.ably.lib.util;
+
+import android.os.Build;
+import io.ably.lib.BuildConfig;
+import io.ably.lib.transport.Defaults;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class AgentHeaderCreator {
+    /**
+     * Separates agent entries from each other.
+     */
+    private static final String AGENT_ENTRY_SEPARATOR = " ";
+
+    /**
+     * Separates agent name from agent version.
+     */
+    private static final String AGENT_DIVIDER = "/";
+    private static final String ANDROID_LIBRARY_NAME = "android";
+
+    public static String create(Map<String, String> additionalAgents) {
+        StringBuilder agentStringBuilder = new StringBuilder();
+        if (!additionalAgents.isEmpty()) {
+            agentStringBuilder.append(getAdditionalAgentEntries(additionalAgents));
+        }
+        agentStringBuilder.append(Defaults.ABLY_AGENT_VERSION);
+        if (BuildConfig.LIBRARY_NAME == ANDROID_LIBRARY_NAME) {
```

Comment:
It's a rather naive check but I didn't have any better idea :thinking:

Reply:
It would be better if you used `.equals()` rather than `==`.

Reply:
Too much Kotlin and I've forgotten how to write in Java :sweat_smile: Thanks for catching that :wink: https://github.com/ably/ably-java/pull/671/commits/ddd279cd03120efbf46b3f76f09e2fec7938bdcb